### PR TITLE
Basic ValidationException tests, extended class slightly

### PR DESCRIPTION
**Pull request by jbrooksuk**

### CODE REVIEW

File: src/ValidationException.php [hunk 1]
```diff
@@ -38,18 +38,38 @@ class ValidationException extends RuntimeException implements MessageProvider
...
-        return $this->errors;
+        return $this->getErrors();
+    }
+
+    /**
+     * Sets the validation errors.
+     *
+     * @param \Illuminate\Support\MessageBag $errors
+     */
+    public function setErrors(MessageBag $errors)
```
Comment (GrahamCampbell) [inline src/ValidationException.php]: We don't want this.

Comment (GrahamCampbell) [inline src/ValidationException.php]: The exception should be immutable.

Comment (jbrooksuk) [inline src/ValidationException.php]: Sure.

Comment (jbrooksuk) [inline src/ValidationException.php]: Done.

**Comment by GrahamCampbell** [comment]
Can we not use mocks please.

**Comment by jbrooksuk** [comment]
> Can we not use mocks please. 

Lol, sure. Quickest way I could think of. Will change when I have some time.

File: src/ValidationException.php [hunk 2]
```diff
@@ -44,12 +44,22 @@ public function __construct(MessageBag $errors)
     }
 
     /**
+     * Get the validation errors.
+     *
+     * @return \Illuminate\Support\MessageBag
+     */
+    public function getErrors()
```
Comment (GrahamCampbell) [inline src/ValidationException.php:51]: I didn't call it getErrors on purpose to be consistent with laravel's contracts. getErrors is an alias so the original method should always be preferred.

**Pull request closed without merging**